src/main/java/uk/ac/ebi/eva/pipeline/io/readers/NonAnnotatedVariantsMongoReader.java

```diff
@@ -31,39 +41,73 @@
  * {@link org.springframework.batch.item.data.MongoItemReader} is using
  * pagination and it is slow with large collections
  */
-public class NonAnnotatedVariantsMongoReader extends MongoDbCursorItemReader {
+public class NonAnnotatedVariantsMongoReader implements ItemStreamReader<VariantWrapper>, InitializingBean {
+
+    private MongoDbCursorItemReader delegateReader;
+
+    private DBObjectToVariantConverter converter;
 
     private static final String STUDY_KEY = VariantToDBObjectConverter.FILES_FIELD + "."
             + VariantSourceEntryToDBObjectConverter.STUDYID_FIELD;
 
     /**
-     * @param studyId Can be the empty string, meaning to bring all non-annotated variants in the collection.
-     *  If the studyId string is not empty, bring only non-annotated variants from that study. This parameter should
-     *  not be null in any case.
+     * @param studyId Can be the empty string, meaning to bring all non-annotated variants in the collection. If the
+     * studyId string is not empty, bring only non-annotated variants from that study. This parameter should not be null
+     * in any case.
      */
     public NonAnnotatedVariantsMongoReader(MongoOperations template, String collectionsVariantsName, String studyId) {
         if (studyId == null) {
             throw new IllegalArgumentException("NonAnnotatedVariantsMongoReader needs a non-null studyId " +
                     "(it can take a studyId or an empty string for reading every study)");
         }
 
-        setTemplate(template);
-        setCollection(collectionsVariantsName);
+        delegateReader = new MongoDbCursorItemReader();
+        delegateReader.setTemplate(template);
+        delegateReader.setCollection(collectionsVariantsName);
 
         BasicDBObjectBuilder queryBuilder = BasicDBObjectBuilder.start();
         if (!studyId.isEmpty()) {
             queryBuilder.add(STUDY_KEY, studyId);
         }
         DBObject query = queryBuilder.add("annot.ct.so", new BasicDBObject("$exists", false)).get();
-        setQuery(query);
+        delegateReader.setQuery(query);
 
-        String[] fields = { "chr", "start", "end", "ref", "alt" };
-        setFields(fields);
+        String[] fields = {"chr", "start", "end", "ref", "alt"};
+        delegateReader.setFields(fields);
+
+        converter = new DBObjectToVariantConverter();
     }
 
     @PostConstruct
     @Override
     public void afterPropertiesSet() throws Exception {
-        super.afterPropertiesSet();
+        delegateReader.afterPropertiesSet();
+    }
+
+    @Override
+    public void open(ExecutionContext executionContext) throws ItemStreamException {
+        delegateReader.open(executionContext);
+    }
+
+    @Override
+    public VariantWrapper read()
+            throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
```

Comment:
I kept them to fit the interface, but now I think about it, it doesn't really add much, I will remove them